### PR TITLE
Update the mac_ios group to include newer dimension values for macOS,…

### PR DESCRIPTION
**Pull request by yusuf-goog**
… xocde and iOS.

This allows us to add a couple updated testbeds to the prod pool, and run jobs on them as well.
Once they are validated, we will move to update all testsbeds in prod and remove the older values
from the .ci.yaml file.

Also removed bringup: True property of Linux ci_yaml flutter roller, to increase testability.

Bug:94767

*Replace this paragraph with a description of what this PR is changing or adding, and why. Consider including before/after screenshots.*

*List which issues are fixed by this PR. You must list at least one issue.*

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: .ci.yaml [hunk 1]
```diff
@@ -106,9 +106,12 @@ platform_properties:
...
         ]
-      os: Mac-10.15
-      device_os: iOS-14.4.2
-      xcode: 12c33
+      # TODO (mohsinally@): Remove the multi-values in 
+      #  dimensions below once the fleet is all upgraded.
+      #  https://github.com/flutter/flutter/issues/94767
+      os: Mac-10.15|Mac-12.0
+      device_os: iOS-14.4.2|iOS-15.1
+      xcode: 12c33|13a233
```
Comment (CaseyHillers) [inline .ci.yaml:114]: This shouldn't have the `|` as it's a test level construct

File: .ci.yaml [hunk 2]
```diff
@@ -248,7 +251,6 @@ targets:
 
   - name: Linux ci_yaml flutter roller
     recipe: infra/ci_yaml
-    bringup: true # TODO(chillers): https://github.com/flutter/flutter/issues/93225
```
Comment (CaseyHillers) [inline .ci.yaml:251]: Thanks!

File: .ci.yaml (hunk 1 above)
Comment (yusuf-goog) [reply .ci.yaml:114]: Should it just be a space?

Comment (godofredoc) [reply .ci.yaml:114]: Does dimension matching support ORs?

File: .ci.yaml [hunk 3]
```diff
@@ -106,9 +106,12 @@ platform_properties:
           {"dependency": "gems"},
           {"dependency": "ios_signing"}
         ]
-      os: Mac-10.15
-      device_os: iOS-14.4.2
-      xcode: 12c33
+      # TODO (mohsinally@): Remove the multi-values in 
+      #  dimensions below once the fleet is all upgraded.
+      #  https://github.com/flutter/flutter/issues/94767
+      os: Mac-10.15 Mac-12.0
```
Comment (godofredoc) [inline .ci.yaml:112]: I don't think swarming APIs allow multi-values. Let's swap all at once after hours.

Comment (yusuf-goog) [reply .ci.yaml]: done

**Pull request merged**